### ISSUE TITLE
How to get upsample opreate type?

### ISSUE THREAD

**Issue by wshj2020** [onnx/onnx#2784]
I export my pth model to onnx，my layer is 'nn.upsample' ,but my op type is resize, how can I get  'upsample' type, thank you!

**Comment by TMVector** [comment]
Hi @wshj2020, are you sure [`Resize`](https://github.com/onnx/onnx/blob/master/docs/Operators.md#Resize) won't do what you want? It looks like it has some of the same functionality as the PyTorch [Upsample](https://pytorch.org/docs/master/generated/torch.nn.Upsample.html) operator.

Have you tried running your ONNX models to see if it computes the same thing as your PyTorch model?

**Comment by wshj2020** [comment]
Thanks for your reply, I konw what you mean, but if I want to get 'upsample' result, what should I do? my code is like this :
     " m = nn.Upsample(scale_factor=2, mode='bilinear',align_corners=False)"
I modify 'bilinear' to 'nearest', I can get 'upsample' operate type, but is there any other better methods?

**Comment by TMVector** [comment]
Can you tell me why you want the `Upsample` operator in particular? Is it because of converter support?

The [`Upsample`](https://github.com/onnx/onnx/blob/master/docs/Operators.md#Upsample) operator was deprecated in opset 10 in favour of the [`Resize`](https://github.com/onnx/onnx/blob/master/docs/Operators.md#Resize) operator.
The Resize operator supports the case you mention (`nn.Upsample(scale_factor=2, mode='bilinear',align_corners=False)`), and if you are having accuracy problems compared to your PyTorch model I can only suggest you open a thread on the [PyTorch discussion forums](https://discuss.pytorch.org/) :)

**Comment by wshj2020** [comment]
yes，my converter only support upsample.

**Comment by TMVector** [comment]
Are you specifying an opset version to the [torch.onnx.export](https://pytorch.org/docs/stable/onnx.html#torch.onnx.export) function? If you're specifying `opset_version=9`, then the exporter shouldn't be producing `Resize` nodes because that operator wasn't introduced until version 10.

**Comment by wshj2020** [comment]
Now, I use nearest mode , I can get "upsample" op_type.

**Comment by sakura-iv** [comment]
you can set onnx_opset_version=9 , then it will be upsample operator

**Comment by omerferhatt** [comment]
> Are you specifying an opset version to the [torch.onnx.export](https://pytorch.org/docs/stable/onnx.html#torch.onnx.export) function? If you're specifying `opset_version=9`, then the exporter shouldn't be producing `Resize` nodes because that operator wasn't introduced until version 10.

It helps to solve `pytorch_half_pixel` error on issue#632 `onnx-tensorflow`
